### ISSUE TITLE
Use 'PaymentCurrencyAmount' Dictionary

### ISSUE THREAD

**Issue by adrianhopebailie** [WICG/webmonetization#16]
From web-monetization created by [adrianhopebailie](https://github.com/adrianhopebailie): adrianhopebailie/web-monetization#11  

The [`PaymentCurrencyAmount`](https://www.w3.org/TR/payment-request/#paymentcurrencyamount-dictionary) dictionary is already defined for exchanging currencies and amounts via Web APIs and is used extensively in the Payment Request API.

We should consider changing the `monetizationprogress` event to use this type instead of `amount`, `asset_code` and `asset_scale`.

**Comment by sidvishnoi** [comment]
What was the rationale for adding `assetScale`? Why `value`/`amount` was not enough? (my guess is floating point errors)

**Comment by sidvishnoi** [comment]
Also, if I remember correctly, WebIDL doesn't allow using a `dictionary` as an interface member. So, we can't really reuse it by replacing assetScale etc. Following would be invalid WebIDL:
```webidl
interface MonetizationProgressEvent : Event {
  constructor(DOMString type, MonetizationProgressEventInit eventInitDict);
  readonly attribute PaymentCurrencyAmount amount; // <- invalid
  readonly attribute DOMString? receipt;
};
```

We can however use it in `MonetizationProgressEventInit` (i.e., only in constructor).

**Comment by marcoscaceres** [comment]
Sid is right in that it a dictionary can't be used for an attribute. I'd be inclined to close this and keep the current API shape.

**Comment by sublimator** [comment]
@marcoscaceres

**Comment by sublimator** [comment]
Would a new "class" have to be created ? i.e. something not a `dictionary` ? Whatever that is

**Comment by marcoscaceres** [comment]
Filed https://github.com/WICG/webmonetization/issues/240 ... we can discuss there.

**Comment by sublimator** [comment]
See: 
https://github.com/WICG/webmonetization/issues/240#issuecomment-1259092434

**Comment by huijing** [comment]
Fixed as part of #193
